### ISSUE TITLE
media-suite_compile.sh : deletes files from 2 packages installed by the suite.

### ISSUE THREAD

**Issue by pat357** [m-ab-s/media-autobuild_suite#1719]
I'm a bit surprised nobody noticed this before, so I thought I should do this.
As usually, I would use a pull request, but I'm completely stuck with that.
I've an old checkout on github for m-ab-s, but I've messed up it.
I just would lke to start fresh, bu**t I can't delete my own messed-up repo** to create a new one.
That's the reason why I can't use pull-request.

Now to the point:
**1. Luajit**
Luajit installs certain files, which get deleted on every run by  the **_clean_old_builds** array.
These files are luajit.exe , luajit-5.1.a, luajit.pc., ...
So on every run of the suite, Luajit will report "files missing" and starts an unneeded rebuilt.
I included a patch below where you can see the details.

**2. Glslang**
Glslang installs files in include/glslang, while the whole dir gets deleted later on due the uninstall from angle.
I added a file from /include/glslang/SPIRV/GlslangToSpv.h to _check and removed this dir from uninstalling angle.

If there is a more efficient way to submit such nits until I figured out how to delete my repo, please let me know.
Would a git diff patch be easier to process ? Any other ideas ?
Now I  just used diff to create the output : I really hope  you can use it....

**3. Patch:**
```
diff -U2 a/media-suite_compile.sh b/media-suite_compile.sh
--- a/media-suite_compile.sh	2020-06-17 09:29:36.000000000 +0200
+++ b/media-suite_compile.sh	2020-06-19 19:02:39.323033300 +0200
@@ -111,14 +111,14 @@
 _clean_old_builds=(j{config,error,morecfg,peglib}.h
     lib{jpeg,nettle,ogg,vorbis{,enc,file},gnurx,regex}.{,l}a
-    lib{opencore-amr{nb,wb},twolame,theora{,enc,dec},caca,magic,luajit-5.1,uchardet}.{l,}a
-    libSDL{,main}.{l,}a libopen{jpwl,mj2,jp2}.{a,pc} lib/lua
+    lib{opencore-amr{nb,wb},twolame,theora{,enc,dec},caca,magic,uchardet}.{l,}a
+    libSDL{,main}.{l,}a libopen{jpwl,mj2,jp2}.{a,pc} 
     include/{nettle,ogg,opencore-amr{nb,wb},theora,cdio,SDL,openjpeg-2.{1,2},luajit-2.0,uchardet,wels}
     regex.h magic.h
-    {nettle,ogg,vorbis{,enc,file},vo-aacenc,sdl,luajit,uchardet}.pc
+    {nettle,ogg,vorbis{,enc,file},vo-aacenc,sdl,uchardet}.pc
     {opencore-amr{nb,wb},twolame,theora{,enc,dec},caca,dcadec,libEGL,openh264}.pc
     libcdio_{cdda,paranoia}.{{l,}a,pc}
     share/aclocal/{ogg,vorbis}.m4
     twolame.h bin-audio/{twolame,cd-paranoia}.exe
-    bin-global/{{file,uchardet}.exe,sdl-config,luajit{,-2.0.4.exe}}
+    bin-global/{{file,uchardet}.exe,sdl-config,luajit-2.0.4.exe}
     libebur128.a ebur128.h
     libopenh264.a
@@ -1825,5 +1825,5 @@
 
 _check=(lib{glslang,OSDependent,HLSL,OGLCompiler,SPVRemapper}.a
-        libSPIRV{,-Tools{,-opt,-link,-reduce}}.a)
+        libSPIRV{,-Tools{,-opt,-link,-reduce}}.a glslang/SPIRV/GlslangToSpv.h)
 if [[ $ffmpeg != no ]] && enabled libglslang &&
     do_vcs "https://github.com/KhronosGroup/glslang.git"; then
@@ -2131,5 +2131,5 @@
         do_simple_print "${orange}mpv will need libGLESv2.dll and libEGL.dll to use gpu-context=angle"'!'
         do_simple_print "You can find these in your browser's installation directory, usually."
-        do_uninstall include/{EGL,GLES{2,3},GLSLANG,KHR,platform} angle_gl.h \
+        do_uninstall include/{EGL,GLES{2,3},KHR,platform} angle_gl.h \
             lib{GLESv2,EGL}.a "${_check[@]}"
         cp -rf include/{EGL,KHR} "$LOCALDESTDIR/include/"

**Comment by 1480c1** [comment]
> I just would lke to start fresh, bu**t I can't delete my own messed-up repo** to create a new one.

Depending on if you know how to use command line git, you could just clone your local, ssh or https, either set m-ab-s as your remote upstream or do a `git fetch https://github.com/m-ab-s/media-autobuild_suite.git` followed by a `git reset --hard FETCH_HEAD` (Assuming you are on your master branch), then a `git push`, that will put your master branch in alignment to the current master branch

Alternatively, if you are using github's ui and as long as you are sure you don't have extra commits in your master branch, you can open a PR on your own fork with the m-ab-s's master as the input branch into your master branch, and then do a rebase merge

**Comment by 1480c1** [comment]
> I included a patch below where you can see the details.

If you do create a patch outside of a PR, please first commit, then do a `git format-patch upstream/master` or whatever the commit right before your commit is, so that way all I would have to do is `git am 0001-patch.patch`

**Comment by pat357** [comment]
Thank you so much for your advise. My current knowledge from the git command line is a bit poor, but I'm willing to learn. It's just that the learning curve is rather steep, and I'm a bit overwhelmed by the number of commands & sub-commands ,...to get something done.
I was able to create a fresh repo and and created a branch for my changes.
Now my pull-request still doesn't look good : it seems I created 2 changes and one has the title "changes to media-suite_compile" (probably default).
I think I know now what I did wrong, so next time I hope it will be perfect.
